### PR TITLE
fix(ui): specify List keys

### LINKED ISSUE #1269: Add/remove items in waiting area is inconsistent
- When adding items to the waiting area, the list is updated inconsistently leading to duplicates.
- Items are being removed from the bottom, instead of the one which is being clicked on.

If the app is refreshed manually, then the list shows the correct items.

|adding|removing|
|-------|---------|
| adding | removing |

Might be due to a missing [keyed each](https://svelte.dev/docs#each).

### PR DESCRIPTION
**Pull request by MeBrei**
Closes #1269

### CODE REVIEW

**Review by rudolfs — APPROVED** [review]

File: ui/DesignSystem/Component/List.svelte [hunk 1]
```diff
@@ -10,6 +10,7 @@
 
   export let dataCy = "";
   export let style = "";
+  export let key = "";
```
Comment (xla) [inline ui/DesignSystem/Component/List.svelte:13]:
```suggestion
  export let key: string | null = null;
```

This should express optionality not default. The caller should be empowered to determine if the items are keyed or not.

File: ui/Screen/Profile/Following.svelte [hunk 2]
```diff
@@ -76,7 +76,7 @@
             exist.
           </p>
         </div>
-        <List items={data.requests} let:item={request}>
+        <List items={data.requests} let:item={request} key="urn">
```
Comment (xla) [inline ui/Screen/Profile/Following.svelte:79]: If a key was passed the items should be keeyed, otherwise not.

**Review by xla — CHANGES_REQUESTED** [review]
Smart little change.

**Review by xla — APPROVED** [review]
🍢 🚒 ⌚️ ❔

**Review by rudolfs — APPROVED** [review]

**Pull request merged**